### DISCUSSION TITLE
DML statements in disguise

### DISCUSSION THREAD

**Discussion by manticore-projects** [dbeaver/dbeaver#21177] [DBeaver Development]
Greetings DBeaver Team,

we have gotten an interesting request: https://github.com/JSQLParser/JSqlParser/issues/1550
```sql
WITH updated_rows AS (
    DELETE FROM documents WHERE something='something'
    RETURNING *
)
SELECT * FROM updated_rows LIMIT 1000;
```

From a previous discussion, I understand that you are interested in filtering DMLs and warn about them before executing.
For the example above, JSQLParser would show a Select/Query (instead of a DML) and so DBeaver likely would miss it.

My suggestion was, to amend each Statement with methods like `isDML()` or `isDDL()` and to look up the true nature of the statement, showing `isDML()==true` for the example above.

Please feel welcome to define your requirements. I am drafting right now and  looking for a most mutual useful API.

**Comment by emironovDB** [comment]
Hello, @manticore-projects ! 
It's a great idea!

Please give us some time to check everything in our part and investigate this issue, and then we come back with more information for you. 
I think it requires for a couple of weeks and in July we will be able to give the answer.

**Comment by LonwoLonwo** [comment]
Hello @manticore-projects 

Thank you for your request! We are always happy to cooperate.

We discussed your proposal and decided that we would accept any option of changes in the library. 
We now have no special requests except those that we have already reported.
